server.go

```diff
@@ -487,46 +487,29 @@ func (app *Yarana) RunPushBatch(user *User) error {
 
 	// Filter Kotos which have no activities in a day
 	// get Activities in parallel
-	activitiesChannel := make(chan []*ActivityData, len(kotos))
+	var mutex = &sync.Mutex{}
 	wg := &sync.WaitGroup{}
+	var pushTargetKotoTitles []string
 	for _, koto := range kotos {
 		wg.Add(1)
-		go func(kotoId string) {
+		go func(kotoId string, kotoTitle string) {
 			acts, _ := app.dataCall.GetActivitiesByKotoDataID(kotoId)
-			activitiesChannel <- acts
-			wg.Done()
-		}(koto.ID)
-	}
-	wg.Wait()
-	close(activitiesChannel)
-	// filter kotos
-	timezone := time.FixedZone(ZONE, TIMEDIFF)
-	var pushTargetKotoTitles []string
-	for actsInOneKoto := range activitiesChannel {
-		// find title of the Koto
-		var kotoTitle string
-		if len(actsInOneKoto) > 0 {
-			for _, koto := range kotos {
-				if koto.ID == actsInOneKoto[0].KotoID {
-					kotoTitle = koto.Title
+			if len(acts) < 1 {
```

Comment:
add comment, "If the koto doesn't have no activities, our bot sends notification."